File: problems/prob4 - Largest palindrome product/sln.go
package main

import (
	"fmt"
	"strconv"
	"math"
)

func main() {
	digit_count := float64(3)
	largest := 0
	for x := 0; x < int(math.Pow(10, digit_count)); x++ {
		for y := 0; y < int(math.Pow(10, digit_count)); y++ {
			product := x * y
			if product > largest && is_palindrome(product) {
				largest = product
			}
		}
	}
	fmt.Println(largest)
}

func is_palindrome(n int) bool {
	arr := []rune(strconv.Itoa(n))
	midpoint := len(arr) / 2
	for i := 0; i < midpoint; i++ {
		if arr[i] != arr[len(arr)-1-i] {
			return false
		}
	}

	return true
}

File: problems/prob4 - Largest palindrome product/sln.java
import java.lang.Math;

class Main {
    public static void main(String[] args) {
        int digit_count = 3;
        int largest = 0;
        for(int x = 0; x < Math.pow(10, digit_count); x++) {
            for(int y = 0; y < Math.pow(10, digit_count); y++) {
                int product = x * y;
                if(product > largest && is_palindrome(product)) {
                    largest = product;
                }
            }
        }
        System.out.println(largest);
    }

    public static boolean is_palindrome(int n) {
        char[] arr = String.valueOf(n).toCharArray();
        int midpoint = arr.length / 2;
        for(int i = 0; i < midpoint; i++) {
            if(arr[i] != arr[arr.length-1-i]) {
                return false;
            }
        }

        return true;
    }
}
